### PR TITLE
Home page list items link to list page

### LINKED ISSUE #33: 16. As a User I want to be able to navigate to the List page when I click a List name on the Home page.
## Summary
When a User clicks on a list name on the `Home` Page, they will be immediately redirected to the `List` Page for that List. This is for easier navigation so user’s do not have to click on a list button to set the list path in local storage and *then* click the List page in the nav bar to navigate to it.

## Acceptance Criteria
- [ ] clicking on a list button on the home page redirects users to the List page displaying that list
- [ ] unnecessary `<p>` tag has been removed from the Home page
     - `<p> Hello from the home (<code>/</code>) page! </p>`
- [ ] if user has no lists the home page displays a message telling the user they have no lists to display
- [ ] “TO-DO” comment is removed from Layout.jsx component to clean up code
- [ ] other console logs removed for cleanliness

## Notes
- Remember to consider accessibility when creating links; as we shouldn't embed links inside buttons(disguising them), maybe making the List Name a Link to allow for styling later is best practice
- Think about how other websites navigate to a list (playlist, cart, etc.) and also consider it could be a link for AT but we could style it (and maybe other links) to look like buttons
- If unsure about removing console logs (and potential other UNNECESSARY comments) ask other team members if they know of other things in other files they might want to remove. !!!Be careful not to remove useful, descriptive comments!!!

### PR DESCRIPTION
**Pull request by sillytsundere**
## Description

The lists on the homepage, when clicked, now not only set the list path but also redirect users to the list page for that list. The homepage also now has conditional rendering to display a message when the user has no lists to display. 

## Related Issue

Closes #33 

## Acceptance Criteria

- [x] clicking on a list button on the home page redirects users to the List page displaying that list
- [x] unnecessary `<p>` tag has been removed from the Home page
     - `<p> Hello from the home (<code>/</code>) page! </p>`
- [x] if user has no lists the home page displays a message telling the user they have no lists to display
- [x] “TO-DO” comment is removed from Layout.jsx component to clean up code
- [x] other console logs removed for cleanliness

## Type of Changes

accessibility, refactoring, enhancement

### Before

Screenshot 2024-03-26 at 1 59 45 PM
Screenshot 2024-03-26 at 2 02 43 PM

### After

Screenshot 2024-03-26 at 1 59 21 PM
Screenshot 2024-03-26 at 2 03 36 PM

## Testing Steps / QA Criteria

- `git pull`
- `git checkout pc-feature-home-list-links`
- `npm ci`
- `npm start`
- navigate to `Home` page
- if there are no lists, note message indicating there are no lists
- if there are lists to display, click on a list button
- you will be redirected to the list page for that list
- if you do not have any lists, note the message stating you do not have any lists on the homepage
- if you do have lists, sign out and navigate to the homepage, note the message stating there are no lists to display

## Notes
This PR does not yes have styling changes that I want to make, such as for the icons on the homepage that link to the Github repo and the Collab Lab website, however I wanted to push these changes and ask a question.

Do you note that there is a delay when you click on a list, get navigated to the list page, then when you click on a different list and are navigated to the list page, there is a slight delay when it displays the items from the previous list? I want to ask if there is some way that I can change that. Will it require nesting the setListPath and navigate functions into promises? I think it has to deal with the list path not QUITE being set fast enough before the page redirect. Please let me know your comments or thoughts on this as well as anything else. 

I will push up more changes with the styling later. 

## Update

Added the logos and links for the repo and The Collab Lab to the homepage and made it so they only render on the homepage and are fixed to the bottom left of the page above the navbar. Refactored the List items on home page to links instead of buttons. 

## Acceptance Criteria
- [x] Added github and collab lab links to home page as icons, 
- [x] fix them to the bottom left of the page

### After
Screenshot 2024-03-28 at 1 22 22 PM
Screenshot 2024-03-28 at 1 22 33 PM

### CODE REVIEW

**Review by rachelspencer — APPROVED** [review]
Hi @sillytsundere!

Nice work, this added functionality makes the app already much nicer to use!

Could we move `StackedLogo.jsx` and `CollabLabLogo.jsx` out of views and into another directory? I think we may have moved them into this directory when we were playing around with the import paths while debugging.

With regard to the slight delays, I experienced this also in my issue when the list ownership renders, specifically if the list is not owned by the current user. I am thinking when our two issues merge, all of this info will appear with a slight delay, and I wonder if implementing a loading state for the `List` page could enhance the user experience? I found [this tutorial ](https://www.youtube.com/watch?v=qR-Cj1dGgkM)on how we can do this, its for Angular, but gives us an idea.

This loading state could also be implemented on the `Manage List` and `Home` pages too.

**Review by jaguilar89 — APPROVED** [review]
@sillytsundere This looks great! Good job. Regarding the delay thing, it's a common issue in React, the go-to solution is usually to add a loading page in between renders.

Radix has 2 components that are worth experimenting with: 
**Progress Bar**: https://www.radix-ui.com/themes/docs/components/progress
**Skeleton Screen**: https://www.radix-ui.com/themes/docs/components/skeleton

File: src/views/Home.jsx [hunk 1]
```diff
@@ -48,16 +48,42 @@ export function Home({ data, setListPath }) {
...
+					))}
+				</ul>
+			) : (
+				<div>
+					<p>
+						There are no lists to display. Please create a list to get started!
+					</p>
+				</div>
+			)}
+			{/* <div className="home-links">
```
Comment (trushmi) [inline src/views/Home.jsx]: Just a nit: will we keep this here? I saw the same code further down. If not, we can also delete 3 import lines from the file:
```
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSquareGithub } from '@fortawesome/free-brands-svg-icons';
import StackedLogo from './StackedLogo.jsx';
```

**Review by trushmi — APPROVED** [review]
Great work with this issue @sillytsundere! Regarding a delay, I would suggest adding a loader to address an issue with a small delay and to provide a smoother experience for the user.

File: src/views/Home.jsx (hunk 1 above)
Comment (sillytsundere) [reply src/views/Home.jsx]: I can delete these! I had kept it while I was working on it.

Comment (sillytsundere) [reply src/views/Home.jsx]: And in response to what Jose has said, I will experiment with the loading bars and convene with Rachel with what we think we want to do about the delays. Also Jose and Iryna we can talk about what we'd like to do about the delays as well. Thanks!

**Review by timothy-taylor — APPROVED** [review]
Working pretty well in my testing! Nice job!

As you surmised, the delay is caused by navigating to the page before the new list data is fully loaded. A loader would probably solve this, but figuring out the correct logic for displaying it isn't as straightforward as it could be in this scenario. I might try manually calling `navigate()` fn at the end of the click handler rather than using a link (not sure if this will 100% work, but its where my mind goes).

**Pull request merged**